### ISSUE TITLE
Add a LayeredExtremum capability

### ISSUE THREAD

**Issue by GiudGiud** [idaholab/moose#24682]
## Reason
The LayeredXXX and NearestXXX concepts are cool templates for user objects and postprocessors, but we still need to instantiate them for all the operations of interest.
We have average and integral, users need extrema too

## Design
Add new objects, tests and so on

## Impact
Simulation postprocessing capability improvement

**Comment by GiudGiud** [comment]
I think we can use the MeshDivisionFunctorReductionVPP to do this now. I d like to see it used for this pupose before I close this

**Comment by gambka** [comment]
@GiudGiud I'm using the `LayeredExtremumMaterialProperty` UserObject for a given problem. I've used a lot of the objects that inherit from LayeredBase in the past. This one is unique is that it also inherits from a Postprocessor `ElementExtremeMaterialProperty` postprocessor which gives the maximum over the whole domain rather than several values over the user defined number of layers. I think the reason inheritance from the postprocessor was done is to remove a lot of the code duplication. However, I'm seeing behavior I don't want. When I have only the userobject in my input I get a postprocessor with the same name in my output csvs.

I understand that postprocessors are a specialization of a userobject but I don't believe any user would expect a postprocessor to be added to their outputs by including a specialized LayeredBase type userobject. As you mention in this issue you have lots of other LayeredIntegral, LayeredAverage, etc. None of those type of layered userobjects inherit from their corresponding postprocessor version. They are distinct because the approach to output information from the UO is via the SpatialUserObjectAux. 

In the meantime, I can simply add the erroneus PP to the hide list for the CSV output but I think this should be fixed. The value it reports in the PP when using the UO is also meaningless.

**Comment by GiudGiud** [comment]
Seems like you have identified a flow in the current design. 
Want to raise an issue? We can probably address this one automatically

**Comment by gambka** [comment]
This is an issue for the LayeredExtremum capability, do you want me to make new one with a MWE?

**Comment by GiudGiud** [comment]
yes I think your issue is wider since all LayeredXYZ that rely on a PP would be affected?

**Comment by gambka** [comment]
None of the other LayeredXYZ UOs rely on the corresponding PP (at least the ones I use don't).
